drivers/hmis/app/graphql/types/base_field.rb

```diff
@@ -19,6 +22,20 @@ def initialize(*args, default_value: nil, **kwargs, &block)
       extension(PaginationWrapperExtension)
     end
 
+    # Field-level authorization
+    # https://graphql-ruby.org/authorization/authorization.html#field-authorization
+    def authorized?(object, args, ctx)
+      # if `permission:` was given, then require the current user to have the specified permissions on the object
+      base_authorized = super(object, args, ctx)
+      if @permissions.any?
+        base_authorized && @permissions.all? do |perm|
+          current_permission_for_context?(ctx, permission: perm, entity: object)
```

Comment:
This uses the data loader to check permissions.

If the user lacks permission, the field will be returned as null. (Its possible to change that behavior to throw an error, i'm on the fence)